File: Array_Pair_Sum_Divisibility_Problem.py
class Solution:
	def canPair(self, nuns, k):
		if len(nums) % 2 != 0:
		    return False
		mp = {}
		for i in nums:
		    y = i % k
		    if mp.get((k - y) % k, 0) != 0:
		        mp[(k - y) % k] -= 1
		    else:
                mp[y] = mp.get(y, 0) + 1
        for key, value in mp.items():
            if value != 0:
                return False
        return True

File: Array_Pair_Sum_Divisibility_Problem.cpp
class Solution {
  public:
    bool canPair(vector<int> nums, int k){
        if(nums.size()%2 != 0){
            return false;
        }
        unordered_map<int, int> mp;
        for(int i : nums){
            int y = i%k;
            if(mp[(k - y)%k] != 0){
                mp[(k - y)%k]--;
            }
            else{
                mp[y]++;
            }
        }
        for(auto it : mp){
            if(it.second != 0){
                return false;
            }
        }
        return true;
    }
};
